src/plugins/intel_cpu/src/nodes/roi_align_rotated.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "roi_align_rotated.h"
+
+#include <openvino/opsets/opset14.hpp>
+
+#include "common/cpu_convert.h"
+#include "openvino/reference/roi_align.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+ROIAlignRotated::ROIAlignRotated(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr context)
+    : Node(op, context, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)) {
+    const auto roiAlign = ov::as_type_ptr<const ov::opset14::ROIAlignRotated>(op);
+    pooledH = roiAlign->get_pooled_h();
+    pooledW = roiAlign->get_pooled_w();
+    spatialScale = roiAlign->get_spatial_scale();
+    samplingRatio = roiAlign->get_sampling_ratio();
+    clockwiseMode = roiAlign->get_clockwise_mode();
+}
+
+void ROIAlignRotated::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset14::ROIAlignRotated.
+}
+
+void ROIAlignRotated::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    ov::element::Type inputPrec0 = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type outputPrec = getOriginalOutputPrecisionAtPort(0);
+
+    addSupportedPrimDesc(
+        {{LayoutType::ncsp, inputPrec0}, {LayoutType::ncsp, ov::element::f32}, {LayoutType::ncsp, ov::element::i32}},
```

Comment:
As per specification, 1st and 2nd inputs have the same precisions. Why the second one is always f32?
Also need to check this value. Plugin supports only f32, f16 and bf16 floating point types.